.github/workflows/tag-and-release.yaml

```diff
@@ -9,10 +9,14 @@ permissions:
   packages: read # Allows reading the content of the repository's packages.
   id-token: write
 
+# Permissions for WORKFLOW_TOKEN: workflow/repo
+
 on:
   push:
     branches:
       - main
+  release:
+    types: [published]
```

Comment:
I don't think we actually want this on release published because the zarf package OCI artifacts won't actually exist immediately on GitHub release creation. Instead if we just run the notify job conditionally (see other suggestion) it will make sure it runs after release publish.